index.mjs

```diff
@@ -0,0 +1,165 @@
+import {URL, pathToFileURL} from 'node:url'
+
+import {unified} from 'unified'
+import {engine} from 'unified-engine'
+import {VFile} from 'vfile'
+import {
+  createConnection,
+  Diagnostic,
+  DiagnosticSeverity,
+  Position,
+  ProposedFeatures,
+  Range,
+  TextDocuments,
+  TextDocumentSyncKind
+} from 'vscode-languageserver/node.js'
+import {TextDocument} from 'vscode-languageserver-textdocument'
+
+/**
+ * Convert a unist point to a language server protocol position.
+ *
+ * @param {import('unist').Point} point
+ * @returns {Position}
+ */
+function unistPointToLSPPosition(point) {
+  return Position.create(point.line - 1, point.column - 1)
+}
+
+/**
+ * Convert a unist position to a language server protocol range.
+ *
+ * If no position is given, a range is returned  which represents the beginning
+ * of the document.
+ *
+ * @param {import('unist').Position?} position
+ * @returns {Range}
+ */
+function unistLocationToLSPRange(position) {
+  if (!position) {
+    return Range.create(0, 0, 0, 0)
+  }
+
+  const start = unistPointToLSPPosition(position.start)
+
+  return Range.create(
+    start,
+    // Fall back to start if the end position contains null values
+    position.end.line && position.end.column
+      ? unistPointToLSPPosition(position.end)
+      : start
+  )
+}
+
+/**
+ * Convert a vfile message to a language server protocol diagnostic.
+ *
+ * @param {import('vfile-message').VFileMessage} message
+ * @param {string|undefined} source
+ * @returns {Diagnostic}
+ */
+function vfileMessageToDiagnostic(message, source) {
+  const diagnostic = Diagnostic.create(
+    unistLocationToLSPRange(message.position),
+    message.reason,
+    message.fatal ? DiagnosticSeverity.Error : DiagnosticSeverity.Warning,
```

Comment:
I imagined these severity levels make sense. Possible levels are: `Error`, `Warning`, `Information`, and `Hint`. Previously `Hint` was used.

Reply:
The proposed Error and Warning levels seem fine to me. :+1:

Reply:
perhaps more apt would be `true` -> `Error`, `false` -> `Warning`, `null|undefined` -> `Information`, in places that support more than errors and warnings.
See:
- https://github.com/vfile/vfile-reporter/blob/a893c994f6e2092c4867bd3940e092eccce99386/lib/index.js#L48-L53
- https://github.com/unifiedjs/unified-engine-atom/blob/605326d08a6d04b66aa8abf3143313f24c507699/index.js#L15-L20